### ISSUE TITLE
Triplet model implementation limiting flexibility and efficiency

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#458]
The current implementation of the triplet model is in TensorFlow, which is limiting its flexibility and efficiency. Additionally, the custom implementation of the triplet loss function may not be optimal, as it does not take advantage of PyTorch's built-in loss functions.
